src/Microsoft.ServiceFabric.Services.Remoting/Description/MethodDescription.cs

```diff
@@ -61,10 +61,11 @@ public MethodInfo MethodInfo
      
         internal static MethodDescription Create(string remotedInterfaceKindName, MethodInfo methodInfo)
         {
-            var argumentList = new List<MethodArgumentDescription>();
+            var parameters = methodInfo.GetParameters();
+            var argumentList = new List<MethodArgumentDescription>(parameters.Length);
             var hasCancellationToken = false;
 
-            foreach (var param in methodInfo.GetParameters())
+            foreach (var param in parameters)
```

Comment:
Looking at the IL, the GetParameters() is done before loop the loop so it gets optimized already.